src/commands/Minion/blastfurnace.ts

```diff
@@ -0,0 +1,181 @@
+import Smithing from '../../lib/skilling/skills/smithing';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { BotCommand } from '../../lib/structures/BotCommand';
+import { ItemBank } from '../../lib/types';
+import { BlastfurnaceActivityTaskOptions } from '../../lib/types/minions';
+import {
+	bankHasItem,
+	formatDuration,
+	itemID,
+	itemNameFromID,
+	removeItemFromBank,
+	skillsMeetRequirements,
+	stringMatches
+} from '../../lib/util';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+
+export function hasSkillRequirements(user: KlasaUser): boolean {
+	return skillsMeetRequirements(user.rawSkills, {
+		crafting: 12,
+		firemaking: 16,
+		magic: 33,
+		mining: 50,
+		smithing: 20,
+		thieving: 13
+	});
+}
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '<quantity:int{1}|name:...string> [name:...string]',
+			aliases: ['bf', 'blastfurnace'],
+			usageDelim: ' ',
+			categoryFlags: ['minion', 'skilling'],
+			description: 'Sends your minion to smelt items, which is turning ores into bars.',
+			examples: ['+Blastfurnace bronze']
+		});
+	}
+
+	@requiresMinion
+	@minionNotBusy
+	async run(msg: KlasaMessage, [quantity, barName = '']: [null | number | string, string]) {
+		if (typeof quantity === 'string') {
+			barName = quantity;
+			quantity = null;
+		}
+
+		const bar = Smithing.BlastableBars.find(
+			bar =>
+				stringMatches(bar.name, barName) || stringMatches(bar.name.split(' ')[0], barName)
+		);
+		const hasSkills = hasSkillRequirements(msg.author);
+		if (!hasSkills) {
+			return msg.send(
+				`You do not have high enough stats to use the Blast Furnace, you need atleast crafting: 12, firemaking: 16, magic: 33, mining: 50, smithing: 20 ,and thieving: 13 `
+			);
+		}
+
+		if (!bar) {
+			return msg.send(
+				`Thats not a valid bar to smelt. Valid bars are ${Smithing.BlastableBars.map(
+					bar => bar.name
+				).join(', ')}.`
+			);
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Smithing) < bar.level) {
+			return msg.send(
+				`${msg.author.minionName} needs ${bar.level} Smithing to smelt ${bar.name}s.`
+			);
+		}
+
+		let timeToSmithSingleBar = bar.timeToUse + Time.Second / 10;
+
+		// check if they have a coal bag
+		let coalbag = '';
+		if (
+			msg.author.hasItemEquippedOrInBank(itemID('Coal bag')) &&
+			(bar.id === itemID('Steel Bar') ||
+				bar.id === itemID('Mithril Bar') ||
+				bar.id === itemID('Adamantite Bar') ||
+				bar.id === itemID('Runite Bar'))
+		) {
+			coalbag = `\n\n**Boosts:** 60% speed boost for coal bag.`;
+			timeToSmithSingleBar *= 0.62;
+		}
+		let graceful = '';
+		if (!hasGracefulEquipped(msg.author.getGear('skilling'))) {
+			timeToSmithSingleBar *= 1.075;
+			graceful = `-7.5% time penalty for not having graceful equipped`;
```

Comment:
add at least one newline or else this is gonna have no whitespace between it and the end of the last message